Resources/public/js/views/services/plugins/ez-objectrelationsloadplugin.js

```diff
@@ -44,23 +51,34 @@ YUI.add('ez-objectrelationsloadplugin', function (Y) {
                 );
 
             Y.Array.each(contentDestinations, function (value) {
-                var actualRelatedContent;
-
                 if (!loadedRelation[value.destination]) {
-                    actualRelatedContent = new Content();
                     loadedRelation[value.destination] = true;
 
-                    actualRelatedContent.set('id', value.destination);
-                    actualRelatedContent.load(loadOptions, stack.add(function (error) {
-                        if (error) {
-                            e.target.set("loadingError", true);
-                            loadingError = true;
-                        } else {
-                            relatedContentListArray.push(actualRelatedContent);
-                        }
-                    }));
+                    if (e.loadLocation || e.loadLocationPath) {
+                        this._loadContentStruct(value.destination, e.loadLocation, e.loadLocationPath, stack.add(
+                            function (error, contentStruct) {
+                                if (error) {
+                                    e.target.set("loadingError", true);
+                                    loadingError = true;
+                                } else {
+                                    relatedContentListArray.push(contentStruct);
+                                }
+                            }
+                        ));
+                    } else {
+                        this._loadContent(value.destination, stack.add(
+                            function (error, content) {
```

Comment:
the very same function is used in both case, you should put it in a local variable.